ai4good/models/cm/functions.py

```diff
@@ -29,6 +29,131 @@ class Simulator:
     def __init__(self, params: Parameters):
         self.params = params
 
+    def ode_system2d(self, t, y,  # state of system
+                   infection_matrix, age_categories, symptomatic_prob, hospital_prob, critical_prob, beta,  # params
+                   latentRate, removalRate, hospRate, deathRateICU, deathRateNoIcu,  # more params
+                   better_hygiene, remove_symptomatic, remove_high_risk, ICU_capacity  # control
+                   ):
+        ##
+        params = self.params
+
+        y2d = y.reshape(age_categories, self.params.number_compartments).T
```

Comment:
Can't we just reverse the order of parameters rather than taking a transpose at the end?

Reply:
this is another optimization. this is hard to validate, i just leave input and out the same as it was. in this case it will be different and it requires to revisit all equations. Also we can solve it by switching to another library which ode accept 2d and in this case we can just get rid of input and output transformations at all.